Some fixes:

luantak opened the pull request:
- build a string instead of printing, to allow for more flexible usage

- export all Struct Fields/ Types to make everything user accessible,
this also enables to encode the data to formats like JSON

- remove the struct definition from the README
(Users should use the automatically generated documentation instead,
because it cannot get outdated)

- GetVersion returns by value to indicate, that it can't be nil

- simplify code

luantak commented on osinfo_test.go:
```diff
@@ -7,14 +7,14 @@ import (
 	fmt.Println("-------------")
-	x.PrintInfo()
+
+	fmt.Println(info)
```
This automatically uses the declared String() method of the release type.

luantak commented on ver_linux.go:
```diff
@@ -20,98 +18,89 @@ import (
-	return true
-}
 
-func readFile(filename string) string {
-	content, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return ""
+	if verf := strings.Split(verf, "="); len(verf) >= 2 {
+		info.Version = verf[1]
+	}
```
Add length checks to not panic

JustinTimperio commented on ver_linux.go:
```diff
@@ -20,98 +18,89 @@ import (
-		}
+	out, err := exec.Command("uname", "-r").Output()
+	if err != nil {
+		return info
```
If it can't get the kernel version, it should still continue on and try to read `/etc/os-release`

luantak replied:
ok

JustinTimperio commented on osinfo.go:
```diff
@@ -10,51 +14,68 @@ type Release struct {
 }
 
-type windowsRelease struct {
+type WindowsRelease struct {
```
Why export this?

JustinTimperio commented on osinfo.go:
```diff
@@ -10,51 +14,68 @@ type Release struct {
-	Distro string
-	Kernel string
-	PkgMng string
+type LinuxRelease struct {
```
Why export this?

JustinTimperio commented on osinfo.go:
```diff
@@ -10,51 +14,68 @@ type Release struct {
-type bsdRelease struct {
-	Kernel string
-	PkgMng string
+type BsdRelease struct {
```
Why export this?

JustinTimperio commented on osinfo.go:
```diff
@@ -10,51 +14,68 @@ type Release struct {
 }
 
-type darwinRelease struct {
+type DarwinRelease struct {
```
Why export this?

JustinTimperio commented on ver_windows.go:
```diff
@@ -40,16 +42,14 @@ func GetVersion() *Release {
 	inf.Name = pname
```
You missed lines 38-42. That's a really nice simplification but we should keep it consistent

JustinTimperio replied:
sorry think you got this and github is just slow

JustinTimperio commented on osinfo.go:
```diff
@@ -10,51 +14,68 @@ type Release struct {
 
-	} else if i.Runtime == "darwin" {
-		fmt.Println("Kernel:", i.osx.Kernel)
+func cleanString(s string) string {
```
This was smart

luantak committed:
Some fixes:
- build a string instead of printing, to allow for more flexible usage

- export all Struct Fields/ Types to make everything user accessible,
this also enables to encode the data to formats like JSON

- remove the struct definition from the README
(Users should use the automatically generated documentation instead,
because it cannot get outdated)

- GetVersion returns by value to indicate, that it can't be nil

- simplify code

JustinTimperio commented:
Great work here @lu4p, thanks for the PR
